stores/sql/sqlite/common.go

```diff
@@ -44,9 +44,26 @@ func applyMigration(ctx context.Context, db *sql.DB, fn func(tx sql.Tx) (bool, e
 		} else if !migrated {
 			return nil
 		}
+
 		// perform foreign key integrity check
-		if err := tx.QueryRow(ctx, "PRAGMA foreign_key_check").Scan(); !errors.Is(err, dsql.ErrNoRows) {
-			return fmt.Errorf("foreign key constraints are not satisfied")
```

Comment:
Couldn't you reuse this?

```go
if err := tx.QueryRow(ctx, "PRAGMA foreign_key_check").Scan(&tableName, &rowID, &foreignKey); err == nil {
    return fmt.Errorf("foreign key constraint violation in table '%s': row %d, foreign key %s", tableName, rowID, foreignKey)
} else if !errors.Is(err, dsql.ErrNoRows) {
    return fmt.Errorf("failed to check for foreign key constraint violations")
}
```

Since you don't loop over all rows anyway and return after the first one.

Reply:
I'd rather collect all violations and return them. I updated it to return on unexpected errors, e.g. failing to scan/iterate, and return all violations